### PR TITLE
fix(ma-builder): display advanced settings for all but dust app run

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

- little refactoring to remove the IIFE and reduce repetition
- fix the bug where advanced settings was only shown for dust app run (needs to be the exact opposite of this)

### CODE REVIEW

File: front/components/assistant_builder/ActionsScreen.tsx [hunk 1]
```diff
@@ -509,6 +509,116 @@ function ActionCard({
...
+  instructions: string | null;
+  updateAction: (args: {
+    actionName: string;
+    actionDescription: string;
+    getNewActionConfig: (
+      old: AssistantBuilderActionConfiguration["configuration"]
+    ) => AssistantBuilderActionConfiguration["configuration"];
+  }) => void;
+  setEdited: (edited: boolean) => void;
+}) {
```
Comment (flvndvd) [inline front/components/assistant_builder/ActionsScreen.tsx:534]: `nit`

Extract this to a dedicated interface.

File: front/components/assistant_builder/ActionsScreen.tsx [hunk 2]
```diff
@@ -509,6 +509,116 @@ function ActionCard({
...
+function ActionConfigEditor({
+  owner,
+  action,
+  dustApps,
+  dataSources,
+  instructions,
+  updateAction,
+  setEdited,
+}: {
+  owner: WorkspaceType;
```
Comment (flvndvd) [inline front/components/assistant_builder/ActionsScreen.tsx:521]:
```suggestion
  owner: LightWorkspaceType;
```

File: front/components/assistant_builder/ActionsScreen.tsx [hunk 3]
```diff
@@ -543,179 +653,64 @@ function ActionEditor({
   ].includes(action.type as any);
 
   const shouldDisplayActionDescription = action.type !== "DUST_APP_RUN";
+  const shouldDisplayAdvancedSettings = action.type !== "DUST_APP_RUN";
```
Comment (flvndvd) [inline front/components/assistant_builder/ActionsScreen.tsx:656]: I guess we could just do one of those?

**Review by flvndvd — APPROVED** [review]
LGTM, left a few comments.

File: front/components/assistant_builder/ActionsScreen.tsx (hunk 3 above)
Comment (fontanierh) [reply front/components/assistant_builder/ActionsScreen.tsx:656]: Yeah I think you're right these 2 will stay aligned 👍

File: front/components/assistant_builder/ActionsScreen.tsx (hunk 2 above)
Comment (fontanierh) [reply front/components/assistant_builder/ActionsScreen.tsx:521]: I actually need the full one here

File: front/components/assistant_builder/ActionsScreen.tsx (hunk 1 above)
Comment (fontanierh) [reply front/components/assistant_builder/ActionsScreen.tsx:534]: Might do, but not here

**Pull request merged**